mesh/meshdb.go

```diff
@@ -275,43 +275,40 @@ func (m *DB) ForBlockInView(view map[types.BlockID]struct{}, layer types.LayerID
 
 // LayerBlockIds retrieves all block ids from a layer by layer index
 func (m *DB) LayerBlockIds(index types.LayerID) ([]types.BlockID, error) {
-	idsBytes, err := m.layers.Get(index.Bytes())
-	if err != nil {
-		return nil, err
-	}
-
-	if len(idsBytes) == 0 {
-		// zero block layer
-		return []types.BlockID{}, nil
+	layerBuf := index.Bytes()
+	zero := false
+	ids := []types.BlockID{}
+	it := m.layers.Find(layerBuf)
+	// TODO(dshulyak) iterator must be able to return an error
+	for it.Next() {
+		if len(it.Key()) == len(layerBuf) {
+			zero = true
+			continue
+		}
+		var id types.BlockID
+		copy(id[:], it.Key()[len(layerBuf):])
```

Comment:
what's the diff bwtn copy() and BytesToInterface here? i'd imagine the latter is stronger typed than former. i am thinking if the data is corrupted somehow.

Reply:
actually, in the previous code, BlockID was always written without BytesToInterface and was unmarshalled using it. it wasn't a problem with xdr because apparently it was noop, but it is not the case with most of the formats.

as for the safety, i don't think that there is anything beneficial in using codec if we have bytes already. db will usually check crc for corruption after loading stuff from disk.